common/functions/src/scalars/function.rs

```diff
@@ -50,7 +50,7 @@ pub trait Function: fmt::Display + Sync + Send + DynClone {
         &self,
         _columns: &ColumnsWithField,
         _input_rows: usize,
-        _func_ctx: FunctionContext,
+        _func_opts: FunctionOptions,
```

Comment:
I think it's better to name it `FunctionContext` or `Context`, which can later be used to place some data that needs to be passed between functions, such as `warnings`, `hints`, etc. And This parameter should be the first parameter.

Reply:
I also think `FunctionOptions` is bad name.  Let's keep it `FunctionContext `

Reply:
ok, let keep `FunctionContext` and make it be the first parameter

Reply:
and another point, should we make `FunctionContext` in this func be `Arc<FunctionContext>`cc @sundy-li @b41sh

Reply:
> and another point, should we make `FunctionContext` in this func be `Arc<FunctionContext>`cc @sundy-li @b41sh

It's trivial.

Reply:
Better to use `Arc<FunctionContext>`, there may be a lot of data later.

Reply:
I do think it's better to use `&mut FunctionContext ` to avoid copy, but it's trivial, we can refactor it later.